app_server/test/controllers/UserController.js

```diff
@@ -83,8 +125,48 @@ lab.experiment('UserController log in tests', function () {
     Router.inject({method: 'POST', url: '/api/users/login',
                    payload: {appId: '123456789',
                              accessToken: 'xyz'}}, function (res) {
-      Code.expect(res.statusCode).to.equal(401);
+      Code.expect(res.statusCode).to.equal(400);
       done();
     });
   });
+
+  lab.test('Valid login', function (done) {
+    facebookServer.start(function (err) {
+      Code.expect(err).to.not.be.null();
+
+      Facebook.FACEBOOK_API_URL = 'http://localhost:8888';
+
+      Router.inject({method: 'POST', url: '/api/users/login',
+                    payload: {appId: '123456789',
+                              accessToken: 'xyz'}}, function (res) {
+        var expected = {
+          alias: 'Bob The Builder',
+          username: '18292522986296117@facebook',
+          platformType: 'facebook',
+          platformId: '18292522986296117'
+        };
+
+        Code.expect(res.statusCode).to.equal(200);
+
+        var body = JSON.parse(res.payload);
+        Code.expect(body.alias).to.equal(expected.alias);
+        Code.expect(body.username).to.equal(expected.username);
+        Code.expect(body.platformType).to.equal(expected.platformType);
+        Code.expect(body.platformId).to.equal(expected.platformId);
+
+        Facebook.FACEBOOK_API_URL = 'https://graph.facebook.com';
+        facebookServer.stop(function () {});
```

Comment:
Should the `function () {}` part be removed, then?
I'm not entirely sure if I'm looking up [the correct documentation](http://hapijs.com/api#serverstopoptions-callback), but the callback appears to be optional, so it may be better to remove it if it's not used.